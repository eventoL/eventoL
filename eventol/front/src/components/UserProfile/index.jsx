import React from 'react';
import PropTypes from 'prop-types';

import {PROFILE_URL} from '../../utils/urls';

import './index.scss';

const UserProfile = ({user: {first_name: firstName, last_name: lastName}}) => (
  <div className="user-profile">
    <a href={PROFILE_URL}>
      <div className="user">
        <div className="name">{`${firstName} ${lastName}`}</div>
        <div className="image">
          <div className="fa fa-user fa-2x" />
        </div>
      </div>
    </a>
  </div>
);

UserProfile.propTypes = {
  user: PropTypes.shape({
    first_name: PropTypes.string,
    last_name: PropTypes.string,
  }).isRequired,
};

export default UserProfile;
