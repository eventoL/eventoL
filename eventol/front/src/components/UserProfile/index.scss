@import '../../styles/index';

@media (min-width: 950px) {
  .user-profile {
    margin-left: auto;

    a {
      text-decoration: none;
    }

    .user {
      @include flex-element;

      .image {
        @include center-element;
        -webkit-transition: border .125s ease;
        border: 3px solid transparent;
        border-radius: 44px;
        box-sizing: border-box;
        display: flex;
        height: 44px;
        margin-left: 10px;
        overflow: hidden;
        transition: border .125s ease;
        vertical-align: middle;
        width: 44px;

        img {
          display: block;
          width: 100%;
        }
      }

      .name {
        @include flex-element;
        @include center-element;
        font-size: 18px;
        font-weight: 300;
        height: 44px;
      }
    }

    .user:hover {
      cursor: pointer;

      .image {
        border: 3px solid $eventol-color;
      }
    }

    .user-profile-menu {
      display: none;
    }
  }
}

@media (max-width: 950px) {
  .user-profile {
    background: $user-profile;
    margin-left: auto;
    margin-right: 8vw;
    opacity: .8;
    z-index: 0;

    a {
      // sass-lint:disable no-important
      float: unset !important;
      // sass-lint:enable no-important
      text-decoration: none;
    }

    .user {
      @include flex-element;

      .image {
        -webkit-transition: border .125s ease;
        align-items: center;
        border: 3px solid transparent;
        border-radius: 44px;
        box-sizing: border-box;
        display: flex;
        height: 44px;
        justify-content: center;
        margin-left: 10px;
        overflow: hidden;
        transition: border .125s ease;
        vertical-align: middle;
        width: 44px;

        img {
          display: block;
          width: 100%;
        }
      }

      .name {
        @include flex-element;
        @include center-element;
        font-size: 14px;
        font-weight: 300;
        height: 44px;
        padding-left: 10px;
      }
    }

    .user:hover {
      background: $eventol-color;
      cursor: pointer;
    }

    .user-profile-narrow-menu {
      display: none;
    }
  }
}
