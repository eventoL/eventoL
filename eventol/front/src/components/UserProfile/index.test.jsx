import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('../../utils/urls', () => ({
  PROFILE_URL: 'PROFILE_URL',
}));

import UserProfile from '.';

describe('UserProfile', () => {
  let user;
  let tree;
  let component;

  const getComponent = () => {
    component = renderer.create(<UserProfile user={user} />);
    return component;
  };

  beforeEach(() => {
    user = {
      first_name: 'first_name',
      last_name: 'last_name',
    };
    component = getComponent();
    tree = component.toJSON();
  });

  describe('Default', () => {
    test('Snapshot', () => {
      expect(tree).toMatchSnapshot();
    });
  });
});
