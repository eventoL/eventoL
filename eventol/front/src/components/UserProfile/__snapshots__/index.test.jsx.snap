// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserProfile Default Snapshot 1`] = `
<div
  className="user-profile"
>
  <a
    href="PROFILE_URL"
  >
    <div
      className="user"
    >
      <div
        className="name"
      >
        first_name last_name
      </div>
      <div
        className="image"
      >
        <div
          className="fa fa-user fa-2x"
        />
      </div>
    </div>
  </a>
</div>
`;
